src/words/environment/WordsEnvironment.java

```diff
@@ -10,13 +12,21 @@
 
 public class WordsEnvironment {
 	private HashMap<String, WordsClass> classes;
-	private HashMap<String, WordsObject> objects;
+	/* gettableObjects is a list of locally scoped symbol tables with the most local 
+	 * scope first in the list, and global scope always being last in the list.
+	 */
+	private LinkedList<HashMap<String, WordsObject>> gettableObjects;
+	/* ungettableObjects contains the objects that have gone out of scope, but should still appear
+	 * on the board, and are still used by action listeners */
+	private HashSet<WordsObject> ungettableObjects;
```

Comment:
Yep. The whole advantage is when it comes to removal, we have a constant time removal.